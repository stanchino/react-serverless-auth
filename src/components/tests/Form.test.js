import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import { Form } from "..";

const mockStore = configureStore();

describe("Form", () => {
    it("matches the snapshot", () => {
        const store = mockStore({ auth: { flash: { error: null, notice: null } } });
        expect(renderer.create(<Provider store={store}><Form form={{ handleSubmit: jest.fn() }}/></Provider>).toJSON()).toMatchSnapshot();
    });

    it("while loading", () => {
        const store = mockStore({ auth: { flash: { error: null, notice: null } }});
        const subject = mount(<Provider store={store}><Form form={{ handleSubmit: jest.fn() }} loading={true} /></Provider>);
        expect(subject.text()).toMatch("Loading...");
    });
});