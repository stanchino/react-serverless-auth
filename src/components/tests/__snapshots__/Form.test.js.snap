// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form matches the snapshot 1`] = `
<form
  className={undefined}
  onSubmit={undefined}
>
  <div
    className="messages"
  >
    <div
      className="flash"
    />
  </div>
</form>
`;
